src/variety/araf.js

```diff
@@ -0,0 +1,171 @@
+//
+//  araf.js
+//
+(function(pidlist, classbase){
+	if(typeof module==='object' && module.exports){module.exports = [pidlist, classbase];}
+	else{ pzpr.classmgr.makeCustom(pidlist, classbase);}
+}(
+['araf'], {
+//---------------------------------------------------------
+// マウス入力系
+MouseEvent:{
+	inputModes : {edit:['number','clear'],play:['border','subline']},
+	mouseinput_auto : function(){
+		if(this.puzzle.playmode){
+			if(this.mousestart || this.mousemove){
+				if(this.btn==='left' && this.isBorderMode()){ this.inputborder();}
+				else{ this.inputQsubLine();}
+			}
+		}
+		else if(this.puzzle.editmode){
+			if(this.mousestart){ this.inputqnum();}
+		}
+	}
+},
+
+//---------------------------------------------------------
+// キーボード入力系
+KeyEvent:{
+	enablemake : true
+},
+
+Cell:{
+	minnum: 0,
+	maxnum: 999
+},
+
+Board:{
+	hasborder : 1,
+
+	cols: 8,
+	rows: 8
+},
+
+AreaRoomGraph:{
+	enabled : true
+},
+
+//---------------------------------------------------------
+// 画像表示系
+
+Graphic:{
+	gridcolor_type : "DLIGHT",
+
+	bordercolor_func : "qans",
+
+	circleratio : [0.45, 0.40],
+
+
+	paint : function(){
+		this.drawBGCells();
+		this.drawDashedGrid();
+
+		this.drawCircledNumbers();
+
+		this.drawBorders();
+		this.drawBorderQsubs();
+
+		this.drawChassis();
+		this.drawTarget();
+	}
+},
+
+//---------------------------------------------------------
+// URLエンコード/デコード処理
+Encode:{
+	decodePzpr : function(type){
+		this.decodeNumber16();
+	},
+	encodePzpr : function(type){
+		this.encodeNumber16();
+	}
+},
+
+//---------------------------------------------------------
+// 正解判定処理実行部
+AnsCheck:{
+	checklist : [
+		"checkLessThan2Num",
+		"checkMoreThan2Num",
```

Comment:
This is just my personal preference, but could you move this check below the `checkBigArea` and `checkSmallArea` lines? IMHO the checks for solutions which are (always) invalid should come before the checks for solutions which are unfinished.

Reply:
Fair

Reply:
While I agree with the general idea, I'm not sure it makes sense here? You can't check area size if there's not exactly two clues in an area. (As I'm writing this, test are failing with the reordered checklist, and I'm not sure how that could be fixed.)

Reply:
I suppose the too large/too small checks could check the areas with two clues only... But is the extra complexity worth it?

Reply:
The checks already ignore areas without exactly two clues.